tests/report/polarion/test.sh

```diff
@@ -7,14 +7,97 @@ rlJournalStart
+        rlAssertGrep '<property name="polarion-custom-plannedin" value="RHEL-9.1.0"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-assignee" value="myassignee"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-poolteam" value="mypoolteam"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-arch" value="x86_64"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-platform" value="myplatform"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-build" value="mybuild"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-sampleimage" value="mysampleimage"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-logs" value="mylogslocation"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-composeid" value="mycomposeid"/>' "xunit.xml"
+
+        # The testcase properties
+        rlAssertGrep '<property name="polarion-testcase-id" value="BASEOS-10913"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-testcase-id" value="BASEOS-10914"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-testcase-id" value="BASEOS-10915"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-testcase-project-id" value="RHELBASEOS"/>' "xunit.xml"
+    rlPhaseEnd
+
+    rlPhaseStartTest 'Test the facts properties'
+        rlRun "tmt run -avr execute report -h polarion --no-upload --use-facts --file xunit.xml 2>&1 >/dev/null | tee output" 2
+        rlAssertGrep '<property name="polarion-custom-hostname" value="' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-arch" value="' "xunit.xml"
+    rlPhaseEnd
+
+    rlPhaseStartTest 'The "None" string should nevery be in a property value'
+        rlRun "tmt run -avr execute report -h polarion --no-upload --project-id '' --planned-in '' --template '' --file xunit.xml 2>&1 >/dev/null | tee output" 2
+        rlAssertNotGrep 'value="None"' "xunit.xml"
+    rlPhaseEnd
+
+    rlPhaseStartTest 'Check the plugin behavior based on setting ENV variables'
+        rlRun "export \
+        TMT_PLUGIN_REPORT_POLARION_PROJECT_ID=myprojectid \
+        TMT_PLUGIN_REPORT_POLARION_TITLE=mytitle \
+        TMT_PLUGIN_REPORT_POLARION_DESCRIPTION=mydesc \
+        TMT_PLUGIN_REPORT_POLARION_TEMPLATE=mytemplate \
+        TMT_PLUGIN_REPORT_POLARION_PLANNED_IN=myplannedin \
+        TMT_PLUGIN_REPORT_POLARION_ASSIGNEE=myassignee \
+        TMT_PLUGIN_REPORT_POLARION_POOL_TEAM=mypoolteam \
+        TMT_PLUGIN_REPORT_POLARION_ARCH=x86_64 \
+        TMT_PLUGIN_REPORT_POLARION_PLATFORM=myplatform \
+        TMT_PLUGIN_REPORT_POLARION_BUILD=mybuild \
+        TMT_PLUGIN_REPORT_POLARION_SAMPLE_IMAGE=mysampleimage \
+        TMT_PLUGIN_REPORT_POLARION_LOGS=mylogslocation \
+        TMT_PLUGIN_REPORT_POLARION_COMPOSE_ID=mycomposeid \
+        "
+
+        # TODO: how to check these?
```

Comment:
I just wanted to test the "false" value of `TMT_PLUGIN_REPORT_POLARION_UPLOAD` (which probably means setting `TMT_PLUGIN_REPORT_POLARION_UPLOAD=''`, but I think we can skip this check.

For the `TMT_PLUGIN_REPORT_POLARION_USE_FACTS` I just wanted to check it is effective, which means the difference between setting `TMT_PLUGIN_REPORT_POLARION_USE_FACTS="yes"` and `TMT_PLUGIN_REPORT_POLARION_USE_FACTS=''`.

Reply:
well, you can set it to true/false with almost anything resembling a true/false value https://click.palletsprojects.com/en/8.1.x/parameters/#parameter-types but the thing is since False is the default you have no way of knowing if it had any effect

the facts are easy, just check existence/nonexistence of the hostname property

Reply:
For flags, please use either `1` or `0`, see the last paragraph in https://tmt.readthedocs.io/en/stable/overview.html#plugin-variables:

> For setting flag-like option, 0 and 1 are the expected value. For example, an interactive mode would be enabled in this run:
> 
> ```
> TMT_PLUGIN_EXECUTE_TMT_INTERACTIVE=1 tmt run ... execute -h tmt ...
> ```